lib/_http_server.js

```diff
@@ -278,7 +279,12 @@ function writeHead(statusCode, reason, obj) {
   if (this[kOutHeaders]) {
     // Slow-case: when progressive API and header fields are passed.
     let k;
-    if (obj) {
+    if (ArrayIsArray(obj)) {
+      for (let n = 0; n < obj.length; n += 2) {
+        k = obj[n + 0];
+        if (k) this.setHeader(k, obj[n + 1]);
```

Comment:
Can we do something better than `setHeader` here? That's quite slow.

Reply:
This is the slow path we don’t want to hit. If you never call setHeader this should not be hit.